clients/src/main/java/org/apache/kafka/common/metrics/stats/Max.java

```diff
@@ -36,10 +36,11 @@ protected void update(Sample sample, MetricConfig config, double value, long now
 
     @Override
     public double combine(List<Sample> samples, MetricConfig config, long now) {
-        double max = Double.NEGATIVE_INFINITY;
-        for (Sample sample : samples)
+        double max = Double.MIN_VALUE;
+        for (Sample sample : samples) {
             max = Math.max(max, sample.value);
-        return max;
+        }
+        return Math.abs(max - Double.MIN_VALUE) < 0.001 ? Double.NaN : max;
```

Comment:
Sorry if this basically seems like code-golfing, but I'm wondering if this would be equivalent and a little more robust?

```suggestion
        return samples.isEmpty() ? Double.NaN : max;
```

Then, we could leave the initial values at negative infinity.

Reply:
No, it doesn't seem like code-golfing. Thanks for the review, I think we made the implementation substantially easier to read